### PR TITLE
introducing CAMP_CUSTOM_CVODE and fixing CVodeCreate invocation for SUNDIALS>=4

### PR DESCRIPTION
**Pull request by slayoo**

### CODE REVIEW

**Comment by slayoo** [comment]
CC: @jcurtis2, @nriemer, @mwest1066, @zdaq12 
Following up on last week's discussion, here's a patch to CAMP enabling compilation with newer SUNDIALS and also adding a flag to disable the usage of custom CVODE API.
This branch is now used in compilation of PyPartMC on Zach's CAMP brach.

**Review by mwest1066 — APPROVED** [review]
LGTM. It would be awesome to get this merged so that PyPartMC can use the main branch of CAMP.

**Review by jcurtis2 — APPROVED** [review]

**Comment by slayoo** [comment]
@mattldawson, please let us know if this would work for you?

File: CMakeLists.txt [hunk 1]
```diff
@@ -123,6 +123,7 @@ set(SUNDIALS_LIBS ${SUNDIALS_NVECSERIAL_LIB} ${SUNDIALS_CVODE_LIB}
   ${SUITE_SPARSE_CONFIG_LIB})
 include_directories(${SUNDIALS_INCLUDE_DIR} ${SUITE_SPARSE_INCLUDE_DIR})
 add_definitions(-DCAMP_USE_SUNDIALS)
+add_definitions(-DCAMP_CUSTOM_CVODE)
```
Comment (mattldawson) [inline CMakeLists.txt:126]: should this be set by a CMake option? something like:
```
option(ENABLE_CUSTOM_CVODE "Enable use of custom CVODE v3.6-alpha" ON)
```
or, how were you planning on disabling this when you build for PartMC?

File: src/camp_common.h [hunk 2]
```diff
@@ -19,7 +19,9 @@
 #ifdef CAMP_USE_SUNDIALS
 #include <cvode/cvode.h>             /* Protoypes for CVODE fcts., consts.  */
 #include <cvode/cvode_direct.h>      /* CVDls interface                     */
-#include <cvode/cvode_impl.h>        /* CVodeMem structure                  */
+#ifdef CAMP_CUSTOM_CVODE
+#  include <cvode/cvode_impl.h>        /* CVodeMem structure                  */
+#endif
```
Comment (mattldawson) [inline src/camp_common.h:24]: why does this particular header file need to be optionally included?

**Review by mattldawson — APPROVED** [review]
Looks good to me! Just a couple minor questions, but I'm fine with merging this in.

File: src/camp_common.h (hunk 2 above)
Comment (slayoo) [reply src/camp_common.h:24]: This header file and the `CVodeMem` type are no longer part of newer SUNDIALS API

image

Comment (mattldawson) [reply src/camp_common.h:24]: ah, ok - makes sense.

**Comment by slayoo** [comment]
Thanks!

**Pull request merged**